functorch/_src/partitioners.py

```diff
@@ -203,21 +202,39 @@ def _size_of(metadata):
         torch.uint8: 1,
         torch.bool: 1,
     }
+    if dtype not in sizes:
+        raise NotImplementedError("Don't know the size of dtype ", dtype)
 
+    return numel * sizes[dtype]
+
+def _size_of(node: fx.Node) -> int:
     def to_size_hint(s):
         if isinstance(s, torch.SymIntNode):
             py_s = s.get_pyobj()
             return py_s.shape_env.size_hint(py_s.expr)
         assert isinstance(s, int)
         return s
 
-    numel = _prod(map(to_size_hint, metadata.shape))
-    dtype = metadata.dtype
-
-    if dtype not in sizes:
-        raise NotImplementedError("Don't know the size of dtype ", dtype)
+    if 'val' in node.meta:
+        val = node.meta['val']
+        if isinstance(val, py_sym_types):
+            return 1
+        elif isinstance(val, (list, tuple)):
+            return sum(_tensor_nbytes(to_size_hint(n.numel()), n.dtype) for n in val if isinstance(n, torch.Tensor))
```

Comment:
Maybe a dumb nit: should we use Pytrees, and/or assert that the node has more uniform structure?

this seems like it might do the wrong thing on Tuple[PySymInt, Tensor] or Tuple[PySymInt, PySymInt] (to be fair maybe none of our operators return those types.. today)

Reply:
Yeah that would make sense eventually.